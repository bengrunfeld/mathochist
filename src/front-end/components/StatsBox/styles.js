import styled from 'styled-components'

const Box = styled.div`
  background: #b7b7b7;
  padding: 10px;
  border: 2px solid black;
`

const BoxData = styled.p`
  font-size: 30px;
  font-weight: bold;
  font-family: Nunito;

  color: ${props => {
    if (props.type === 'score') return 'darkred'
    if (props.type === 'time') return 'darkblue'
    return 'black'
  }};
`

export { Box, BoxData }
