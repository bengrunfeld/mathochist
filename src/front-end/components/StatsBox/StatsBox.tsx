import React from 'react'
import PropTypes from 'prop-types'

import { Box, BoxData } from './styles'

const StatsBox = ({ data }) => (
  <Box>
    <BoxData>{data}</BoxData>
  </Box>
)

StatsBox.defaultProps = {
  data: 0
}

StatsBox.propTypes = {
  data: PropTypes.number
}

export default StatsBox
