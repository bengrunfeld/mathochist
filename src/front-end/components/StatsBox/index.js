import StatsBox from './StatsBox'

export default StatsBox
