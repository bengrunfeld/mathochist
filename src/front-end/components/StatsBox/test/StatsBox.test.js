import React from 'react'
import Enzyme, { shallow, mount, render } from 'enzyme'
import { MemoryRouter } from 'react-router-dom'

import StatsBox from '../StatsBox'

describe('<StatsBox />', () => {
  test('matches the snapshot', () => {
    const tree = mount(<StatsBox />)
    expect(tree).toMatchSnapshot()
  })
})
