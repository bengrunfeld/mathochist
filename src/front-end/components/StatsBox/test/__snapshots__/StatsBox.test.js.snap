// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StatsBox /> matches the snapshot 1`] = `
<StatsBox
  data="0"
  statsType="defaultStatsType"
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "iKOMUo",
            "rules": Array [
              "
  display: inline-block;
  background: #b7b7b7;
  padding: 6px;
  border: 2px solid black;
  min-width: 70px;
  border-radius: 5px;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-bdVaJa iKOMUo"
      >
        <styled.p
          type="defaultStatsType"
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": false,
                  "lastClassName": "hZcyxd",
                  "rules": Array [
                    "
  font-size: 30px;
  font-weight: bold;
  font-family: Nunito;
  margin: 0;
  padding: 0;
  text-align: center;

  color: ",
                    [Function],
                    ";
",
                  ],
                },
                "displayName": "styled.p",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "p",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            type="defaultStatsType"
          >
            <p
              className="sc-bwzfXH hZcyxd"
              type="defaultStatsType"
            >
              0
            </p>
          </StyledComponent>
        </styled.p>
      </div>
    </StyledComponent>
  </styled.div>
</StatsBox>
`;
